engine/script.py

```diff
@@ -24,183 +29,54 @@ def compare(path1, path2):
         else:
             return False
 
-def run_c(f, time, mem, input_file, temp_output_file, output_file):
-    compilation = "gcc -Wno-deprecated {} -o compiled_code 2> compile_log".format(f)
+def run_program(command,compilation, time, mem, input_file, temp_output_file, output_file):
     os.system(compilation)
     if (os.stat("compile_log").st_size != 0):
         with open("compile_log", "r+") as temp_file:
-            return {  # Compilation Error
-                "code": 1,
-                "message": temp_file.read()
-            }
+            return code_message(1,temp_file.read())
     else:
-        command = "sudo {} --cpu {} --mem {} --usage usage.txt --exec compiled_code < {} > {}".format(ENGINE_PATH, time, mem, input_file, temp_output_file)
         os.system(command)
-        stat = status()
-        res = None
+        stat=status()
+        res= None
         if stat['run_status'] == "OK":
             if (compare(output_file, temp_output_file)):
                 stat['run_status'] = "AC"
-                res = {  # Passed
-                    "code": 0,
-                    "status": stat
-                }
+                res = code_status(0,stat)
             else:
                 stat['run_status'] = "WA"
-                res = {  # Failed
-                    "code": 0,
-                    "status": stat
-                }
+                res =code_status(0,stat)
         else:
-            res = {
-                "code" : 2,
-                "status" : stat
-            }
+            res= code_status(2,stat)
         os.remove(temp_output_file)
         return res
 
+def run_c(f, time, mem, input_file, temp_output_file, output_file):
+    compilation = "gcc -Wno-deprecated {} -o compiled_code 2> compile_log".format(f)     
+    command = "sudo {} --cpu {} --mem {} --usage usage.txt --exec compiled_code < {} > {}".format(
+            ENGINE_PATH, time, mem, input_file, temp_output_file)
+    return run_program(command,compilation,time, mem, input_file, temp_output_file, output_file) 
```

Comment:
This return statement is being repeated multiple times. Make it DRY

Reply:
![image](https://user-images.githubusercontent.com/56390846/91666645-1a3c6a00-eb1c-11ea-84d5-4054c1ad1e04.png)

Reply:
i didn't understand how would i change the return statement in this

Reply:
You don't need to return a function at all. 
You are free to change other files if requires. 
Follow Object-oriented style.

For ex. You can take the language part as parameter to determine which compilation command to use.
It will be much better if you can make it configurable during setup, then this script will be language agnostic.

Reply:
okay understood